### ISSUE TITLE
Getting multiple outputs from a single layer for custom loss function

### ISSUE THREAD

**Issue by kracwarlock** [fchollet/keras#2082]
I have a layer in a Graph() whose get_output() returns 3 values ->

```
def get_output(.....):
    return (output1, output2, output3)

where output1 = softmax (nb_samples, nb_timesteps, softmax_dim)
      output2 = custom regularization term (nb_samples)
      output3 = custom regularization term (nb_samples)
```

I tried to return three outputs as a tuple with a custom objective function but that is not allowed.

```
def custom_lossfn(output,target):
    softmax = output[0]
    .......
    return T.nnet.categorical_crossentropy(softmax_scaled_clipped) + output[1].sum() + output[2].sum()
```

How can I do this in Keras?

**Comment by EderSantana** [comment]
If you have multiple outputs, you can use `keras.layers.core.SiamseHead` to pick specific outputs.
```python
# get's the first output of the three
graph.add_node(SiameseHead(0), name='Y0', input='output', create_output=True)
graph.add_node(SiameseHead(1), name='Y1', input='output', create_output=True)
graph.add_node(SiameseHead(2), name='Y1', input='output', create_output=True)

```

But since your other outputs are regularization, maybe you want to add them to the layer's `self.regularizers` list instead of using them as output. Both ways the regularization will be part of the cost function, but with `self.regularizers` you don't have to worry about setting up new cost functions. You can calculate the regularization inside `build`, even it has to call `get_output`.

**Comment by kracwarlock** [comment]
And the previous layer's get output method should still return the outputs as a tuple?
```
def get_output(.....):
    return (output1, output2, output3)
```
I don't have access to a machine right now but will definitely check this out on Monday and will close the issue if this works.

**Comment by EderSantana** [comment]
if you add your regularizers to `self.regularizers` you don't have to output them

**Comment by kracwarlock** [comment]
@EderSantana when I return the 3 output tuple with SiameseHead layers it still gives the ndim error since tuple outputs are not allowed.

The last two outputs are not weight regularization terms. They are of shape `(nb_samples_in_batch,dimension)`. Since my batch size is variable I have not been able to add it to `self.regularizers`.

**Comment by kracwarlock** [comment]
I think the issue which you raised here is very much related: https://github.com/fchollet/keras/issues/689
